### PR TITLE
Making the URL parameter optional, modifying exception handling to ignore optional cases.

### PR DESCRIPTION
**Pull request by chamindias**
Edited the ReadMe file as well.

### CODE REVIEW

File: src/main/java/apim/restful/importexport/utils/APIImportUtil.java [hunk 1]
```diff
@@ -180,100 +180,91 @@ public static String extractArchive(File sourceFile, String destination) throws
...
+
+            FileInputStream inputStream = null;
+            BufferedReader bufferedReader = null;
+
+            try {
+                inputStream = new FileInputStream(pathToArchive + APIImportExportConstants.JSON_FILE_LOCATION);
+                bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+                importedApi = new Gson().fromJson(bufferedReader, API.class);
+            } catch (FileNotFoundException e) {
+                log.error("Error in locating api.json file. ", e);
```
Comment (harsha89) [inline src/main/java/apim/restful/importexport/utils/APIImportUtil.java:201]: Better practice would be

``` java
String msg = "Error in locating api.json file"
log.error(msg, e);
throw new APIImportException(msg + e.getMessage());
```

File: src/main/java/apim/restful/importexport/utils/APIImportUtil.java [hunk 2]
```diff
@@ -180,100 +180,91 @@ public static String extractArchive(File sourceFile, String destination) throws
...
-        Set<Tier> allowedTiers = provider.getTiers();
-        boolean isAllTiersAvailable = allowedTiers.containsAll(importedApi.getAvailableTiers());
+        Set<Tier> allowedTiers;
+        Set<Tier> unsupportedTiersList;
 
-        if (!isAllTiersAvailable) {
+        try{
+            allowedTiers = provider.getTiers();
+        } catch (APIManagementException e) {
+            log.error("Error in retrieving tiers of the provider. ", e);
```
Comment (harsha89) [inline src/main/java/apim/restful/importexport/utils/APIImportUtil.java:233]: Same as above

File: src/main/java/apim/restful/importexport/utils/APIImportUtil.java [hunk 3]
```diff
@@ -505,23 +503,22 @@ private static void addAPIWsdl(String pathToArchive, API importedApi, String cur
...
+
         try {
             String swaggerContent = FileUtils.readFileToString(
                     new File(archivePath + APIImportExportConstants.SWAGGER_DEFINITION_LOCATION));
             provider.saveSwagger20Definition(apiId, swaggerContent);
         } catch (APIManagementException e) {
             log.error("Error in adding Swagger definition for the API. ", e);
-            throw new APIImportException("Error in adding Swagger definition for the API. " +
-                    e.getMessage());
+            throw new APIImportException("Error in adding Swagger definition for the API. " + e.getMessage());
```
Comment (harsha89) [inline src/main/java/apim/restful/importexport/utils/APIImportUtil.java:513]: throwiing the exception only with the exception message. Any reason whole exception not throwing

File: src/main/java/apim/restful/importexport/utils/APIImportUtil.java (hunk 1 above)
Comment (chamindias) [inline src/main/java/apim/restful/importexport/utils/APIImportUtil.java:201]: Has been fixed in commit : https://github.com/wso2/product-apim/commit/995f737a32616db5a1ef8950a5d4618f060b530f

File: src/main/java/apim/restful/importexport/utils/APIImportUtil.java (hunk 2 above)
Comment (chamindias) [inline src/main/java/apim/restful/importexport/utils/APIImportUtil.java:233]: Has been fixed in commit : https://github.com/wso2/product-apim/commit/995f737a32616db5a1ef8950a5d4618f060b530f

File: src/main/java/apim/restful/importexport/utils/APIImportUtil.java (hunk 3 above)
Comment (chamindias) [inline src/main/java/apim/restful/importexport/utils/APIImportUtil.java:513]: Has been fixed in commit : https://github.com/wso2/product-apim/commit/995f737a32616db5a1ef8950a5d4618f060b530f

**Pull request merged**